cluster/ec2_banzaicloud.go

```diff
@@ -161,6 +161,23 @@ func (c *EC2ClusterBanzaiCloudDistribution) Persist(string, string) error {
 }
 
 func (c *EC2ClusterBanzaiCloudDistribution) UpdateStatus(status, statusMessage string) error {
+	if c.model.Cluster.ID != 0 && c.model.Cluster.Status != status {
+		statusHistory := &cluster.StatusHistoryModel{
+			ClusterID:   c.model.Cluster.ID,
+			ClusterName: c.model.Cluster.Name,
+
+			FromStatus:        c.model.Cluster.Status,
+			FromStatusMessage: c.model.Cluster.StatusMessage,
+			ToStatus:          status,
+			ToStatusMessage:   statusMessage,
+		}
+
+		err := c.db.Save(&statusHistory).Error
```

Comment:
Can't this be saved via `err := c.db.Save(&c.model)` ? What we need is to have a field in Cluster model that collects status history model entries related to the cluster.

Reply:
See my other comments.